### PR TITLE
FEATURE: add gen_u5_from_qx

### PR DESCRIPTION
**Pull request by krpaulson**
## Describe changes

Add function `gen_u5_from_qx` to calculate <1 and 1-4 ax from qx. The existing function only solved mx --> ax.

## What issues are related

None.

### Packages Repositories

* [x] Have you read the [contributing guidelines](https://github.com/ihmeuw-demographics/packageTemplate/wiki#guide-to-r-package-development) for `ihmeuw-demographics` R packages?
* [x] Have you successfully run `devtools::check()` locally?
* [x] Have you updated or added function (and vignette if applicable) documentation? Did you update the 'man' and 'NAMESPACE' files with `devtools::document()`?
* [x] Have you added in tests for the changes included in the PR?
* [x] Do the changes follow the `ihmeuw-demographics` [code style](https://github.com/ihmeuw-demographics/packageTemplate/wiki/Code-style-guide)?
* [ ] Do the changes need to be immediately included in a new build of [`docker-base`](https://github.com/ihmeuw-demographics/docker-base) or [`docker-internal`](https://github.com/ihmeuw-demographics/docker-internal)? If so follow directions in those repositories to rebuild and redeploy the images.
* [ ] Do the changes require updates to other repositories which use this package? If yes, make the necessary updates in those repos, and consider integration tests for those repositories.

## Details of PR

This came up while developing `model_lifetable`, because we get qx from the MLT steps and need to get ax and mx from there.

### CODE REVIEW

File: R/gen_u5_ax.R [hunk 1]
```diff
@@ -116,3 +93,194 @@ gen_u5_ax <- function(dt, id_cols) {
...
+
+  # settings
+  if (unique(dt$sex) == "male") {
+    # high/low mx threshold coverted to qx
+    threshold <- mx_ax_to_qx(mx = 0.107, ax = 0.330, age_length = 1)
+    # ax value at that high/low threshold
+    baseline_inf <- 0.330
+    baseline_ch <- 1.352
+    # slope/intercept
+    A_inf <- 0.045
```
Comment (chacalle) [inline R/gen_u5_ax.R:208]: I think it came up in code review that this table should maybe be part of `data`

File: R/gen_u5_ax.R [hunk 2]
```diff
@@ -116,3 +93,194 @@ gen_u5_ax <- function(dt, id_cols) {
...
+}
+
+
+
+# ==========================================================================
+#' @title Helper solve function for `gen_u5_ax_from_qx`
+#' @description Function which solves for zero. Uses relationship between
+#'   under-5 mx and ax as described in [gen_u5_ax()].
+#' @param dt \[`data.table()`\] A single life table.
+solve_u5_ax_from_qx <- function(dt) {
```
Comment (chacalle) [inline R/gen_u5_ax.R:196]: Is this just basically transforming the table rules from mx and ax space to qx and ax space?

**Review by chacalle — APPROVED** [review]

File: R/gen_u5_ax.R (hunk 2 above)
Comment (krpaulson) [reply R/gen_u5_ax.R:196]: Yes

File: R/gen_u5_ax.R (hunk 1 above)
Comment (krpaulson) [reply R/gen_u5_ax.R:208]: https://github.com/ihmeuw-demographics/demCore/issues/65

**Comment by krpaulson** [comment]
This is failing checks for `agg_lt` but I'm going to ignore that assuming it will be resolved by PR #64

**Review by chacalle — APPROVED** [review]

**Pull request merged**